### PR TITLE
feat: Metrics Get All Tags API 

### PR DESCRIPTION
**Pull request by Macbeth98**
# Description
- Get All Tag Keys and values Set for the given Metric Name over the time range.

# Testing
- Tested the code by calling the API Endpoint through Postman.

# Checklist:

- [x] I have self-reviewed this PR.
- [x] I have removed all print-debugging and commented-out code that should not be merged.
- [x] I have added sufficient comments in my code, particularly in hard-to-understand areas.
- [x] I have formatted the code, if applicable. For Go, I have run `goimports -w .`.

### CODE REVIEW

File: pkg/segment/results/mresults/metricresults.go [hunk 1]
```diff
@@ -375,6 +375,36 @@ func (r *MetricsResult) GetResultsPromQl(mQuery *structs.MetricsQuery, pqlQueryt
...
+	tagKeyValueSet := make([]string, 0)
+
+	for _, series := range res.AllSeries {
+		tagValues := strings.Split(series.grpID.String(), tsidtracker.TAG_VALUE_DELIMITER_STR)
+		if len(uniqueTagKeys) != len(tagValues)-1 {
+			err := fmt.Errorf("GetResults: the length of tag key and tag value pair must match. UniqueTagKeys length: %v,  TagValues Length: %v", len(uniqueTagKeys), len(tagValues)-1)
+			return nil, nil, err
+		}
+		for index, val := range tagValues[:len(tagValues)-1] {
+			tagKeyValueSet = append(tagKeyValueSet, fmt.Sprintf("%s:%s", uniqueTagKeys[index], val))
```
Comment (Macbeth98) [inline pkg/segment/results/mresults/metricresults.go:401]: - Should I return here? Or Log the Error and continue to process the next series?

- Also, wouldn't there be a case or scenario where the Unique Tag Keys for the Metric may not match with the Tag Values?
   - I meant to say that for a Metric (`Metric_1`), some values may include two tag keys and values. For example: `method: post, respCode: 200`. But some may only include `method: post` with no `respCode`. In this scenario , the `uniqueTagKeysMap` will have both `method` and `respCode`, but for some series, there will not be value for `respCode`.
   - Or is there a general assumption that the above scenario won't happen. Meaning for a given Metric, the tag set will remain constant?

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 2]
```diff
@@ -390,10 +392,36 @@ func ProcessGetAllMetricTagsRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+
+	searchText := fmt.Sprintf("(%v)", metricName)
+
+	metricQueryRequest, _, _, err := convertPqlToMetricsQuery(searchText, timeRange.StartEpochSec, timeRange.EndEpochSec, myid)
+	if err != nil {
+		utils.SendError(ctx, "Failed to parse the Metric Name as a Query", fmt.Sprintf("Metric Name: %+v; qid: %v", metricName, qid), err)
+		return
+	}
+
+	metricQueryRequest[0].MetricsQuery.ExitAfterRawSearch = true
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:408]: nit: maybe rename this field to `ExitAfterTagsSearch`

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 3]
```diff
@@ -390,10 +392,36 @@ func ProcessGetAllMetricTagsRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+
+	segment.LogMetricsQuery("PromQL metrics query parser", &metricQueryRequest[0], qid)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:410]: nit: log as "Tags Request PromQL...`

File: pkg/integrations/prometheus/promql/metricsSearchHandler.go [hunk 4]
```diff
@@ -390,10 +392,36 @@ func ProcessGetAllMetricTagsRequest(ctx *fasthttp.RequestCtx, myid uint64) {
...
+	res := segment.ExecuteMetricsQuery(&metricQueryRequest[0].MetricsQuery, &metricQueryRequest[0].TimeRange, qid)
+
+	uinqueTagKeys, tagKeyValueSet, err := res.GetMetricTagsResultSet(&metricQueryRequest[0].MetricsQuery)
```
Comment (nkunal) [inline pkg/integrations/prometheus/promql/metricsSearchHandler.go:413]: typo in the name of the uniqueTag.. var

File: pkg/segment/results/mresults/metricresults.go [hunk 5]
```diff
@@ -375,6 +375,36 @@ func (r *MetricsResult) GetResultsPromQl(mQuery *structs.MetricsQuery, pqlQueryt
...
+			uniqueTagKeys = append(uniqueTagKeys, tag.TagKey)
+		}
+	}
+
+	tagKeyValueSet := make([]string, 0)
+
+	for _, series := range res.AllSeries {
+		tagValues := strings.Split(series.grpID.String(), tsidtracker.TAG_VALUE_DELIMITER_STR)
+		if len(uniqueTagKeys) != len(tagValues)-1 {
+			err := fmt.Errorf("GetResults: the length of tag key and tag value pair must match. UniqueTagKeys length: %v,  TagValues Length: %v", len(uniqueTagKeys), len(tagValues)-1)
```
Comment (nkunal) [inline pkg/segment/results/mresults/metricresults.go:397]: func name in log is not right

**Review by nkunal — DISMISSED** [review]

**Review by nkunal — APPROVED** [review]

**Pull request merged**